tests/model_fields/models.py

```diff
@@ -156,6 +157,10 @@ class NullDurationModel(models.Model):
     field = models.DurationField(null=True)
 
 
+class DefaultDurationModel(models.Model):
+    field = models.DurationField(default=timedelta(minutes=5))
+
+
```

Comment:
This is not a regression test because it works without the patch. You should try to add `DurationField` to the existing model, e.g.:
```diff
diff --git a/tests/schema/tests.py b/tests/schema/tests.py
index e94659544a..20e85b3c48 100644
--- a/tests/schema/tests.py
+++ b/tests/schema/tests.py
@@ -631,6 +631,18 @@ class SchemaTests(TransactionTestCase):
         # these two types.
         self.assertIn(columns['bits'][0], ("BinaryField", "TextField"))
 
+    def test_add_field_durationfield(self):
+        # Create the table
+        with connection.schema_editor() as editor:
+            editor.create_model(Author)
+        # Add the new field
+        new_field = DurationField(default=datetime.timedelta(minutes=5))
+        new_field.set_attributes_from_name('duration')
+        with connection.schema_editor() as editor:
+            editor.add_field(Author, new_field)
+        columns = self.column_classes(Author)
+        self.assertEqual(columns['duration'][0], 'DurationField')
+
     @unittest.skipUnless(connection.vendor == 'mysql', "MySQL specific")
     def test_add_binaryfield_mediumblob(self):
         """

```

Reply:
that's interesting, because that's how I spotted the issue in my environment.
There must be something I'm missing in how django tests are working.
I thought that the model I created would be applied during tests and would have shown the issue.

I'll definitely implement your approach though.
Thanks for the hint.

Reply:
You probably added `DurationField` to your model or add a `default` and change the `null` flag. In both cases you will hit the issue described in the ticket. Adding a new model to tests is not enough because it will use `CREATE TABLE` statement and ignore a `default` value.

Reply:
gotcha, fair point.

Reply:
With a proper regression test we can revert changes in the `tests/model_fields/models.py`.

Reply:
Wouldn't be useful to keep it to catch potential issues in the creation step, if introduced?

Reply:
Extra test coverage is always welcome, but we cannot mix it with a regression test, it can be added in a separate commit/PR. Moreover tests should be more explicit (have assertions etc.) it's not enough to add a model, unused in test cases, and base the test on the fact that the entire test suite won't crash.

Reply:
I think the issue here is that I'm not familiar with the test environment of the django project and I don't know how to test the creation of a specific model in isolation without breaking the whole test suite.
Unfortunately I don't have time to read the source code to find it out now, but I will in the future.
I'd appreciate if you have any pointers for me for that.
In the meanwhile I'll remove that bit and I'll eventually submit another PR another day to increase the coverage.